### PR TITLE
Osd 8474 orig - e2e  test to delete the rh-api lb on a live cluster and ensure that the lb is recreated in GCP

### PR DESCRIPTION
**Pull request by ritmun**
jira https://issues.redhat.com/browse/OSD-8474 

Following [OSD-8431](https://issues.redhat.com/browse/OSD-8431), tests need to be added to ensure the code logic is correct.

Done Criteria:
 
e2e test to delete the rh-api lb on a live cluster and ensure that the lb is recreated in GCP

### CODE REVIEW

**Comment by MrSantamaria** [comment]
Making changes to Pr-Check:
https://github.com/openshift/release/pull/27263

**Review by MrSantamaria — CHANGES_REQUESTED** [review]

File: pkg/e2e/operators/cloudingress/rhapi_lb.go [hunk 1]
```diff
@@ -77,23 +108,154 @@ func testLBDeletion(h *helper.H) {
...
 						return true, nil
 					}
 					// the rh-api svc hasn't been deleted yet
+					log.Printf("rh-api service not delted yet")
 					return false, nil
 				})
 				Expect(err).NotTo(HaveOccurred())
-			}
-		})
+			}, 600)
```
[the remark this answers is not in the document]
Comment (ritmun) [reply pkg/e2e/operators/cloudingress/rhapi_lb.go:131]: changing 10 min to 15 min

File: pkg/e2e/operators/cloudingress/rhapi_lb.go [hunk 2]
```diff
@@ -77,23 +108,154 @@ func testLBDeletion(h *helper.H) {
...
+					if newLB.Name != oldLB.Name {
+						// A new LB was successfully recreated in GCP
+						return true, nil
+					}
+					// rh-api lb hasn't been deleted yet
+					log.Printf("Old forwarding rule not deleted yet...")
+					return false, nil
+				})
+				Expect(err).NotTo(HaveOccurred())
+			}, 600)
```
[the remark this answers is not in the document]
Comment (ritmun) [reply pkg/e2e/operators/cloudingress/rhapi_lb.go:258]: In my individual test it take 3-4 min max, but I changed it to 15 min, just in case.

**Comment by cblecker** [comment]
@ritmun Would you also mind updating the PR title and body with a description of what this is doing? Thank you!

**Comment by MrSantamaria** [comment]
@ritmun the timeout still seems to be the problem. Could you up this even more?

**Comment by ritmun** [comment]
> @ritmun Would you also mind updating the PR title and body with a description of what this is doing? Thank you!

done

**Comment by MrSantamaria** [comment]
@ritmun Ran it again and digging through the logs:
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_osde2e/1078/pull-ci-openshift-osde2e-main-verify/1509651900734115840

SSSSSSSSSS[restful] 2022/03/31 22:18:39 log.go:30: Old LB - adae1a033fe84440585e25fec3a8f3f2 
[restful] 2022/03/31 22:18:39 log.go:30: Old LB deleted 

Then in Cloudtrail:
DeleteLoadBalancer | March 31, 2022, 16:18:39 (UTC-06:00) | sd-cicd | elasticloadbalancing.amazonaws.com | AWS::ElasticLoadBalancing::LoadBalancer | adae1a033fe84440585e25fec3a8f3f2

It seems the command to create the newLB is never launching.

**Comment by ritmun** [comment]
> @ritmun Ran it again and digging through the logs: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_osde2e/1078/pull-ci-openshift-osde2e-main-verify/1509651900734115840
> 
> SSSSSSSSSS[restful] 2022/03/31 22:18:39 log.go:30: Old LB - adae1a033fe84440585e25fec3a8f3f2 [restful] 2022/03/31 22:18:39 log.go:30: Old LB deleted
> 
> Then in Cloudtrail: DeleteLoadBalancer | March 31, 2022, 16:18:39 (UTC-06:00) | sd-cicd | elasticloadbalancing.amazonaws.com | AWS::ElasticLoadBalancing::LoadBalancer | adae1a033fe84440585e25fec3a8f3f2
> 
> It seems the command to create the newLB is never launching.

Looks like the service deletion never finishes. Maybe that service is in use. Are we sure no other tests are running in parallel on this cluster at the same time? I'll add some logs to debug.

**Comment by ritmun** [comment]
> > @ritmun Ran it again and digging through the logs: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_osde2e/1078/pull-ci-openshift-osde2e-main-verify/1509651900734115840
> > SSSSSSSSSS[restful] 2022/03/31 22:18:39 log.go:30: Old LB - adae1a033fe84440585e25fec3a8f3f2 [restful] 2022/03/31 22:18:39 log.go:30: Old LB deleted
> > Then in Cloudtrail: DeleteLoadBalancer | March 31, 2022, 16:18:39 (UTC-06:00) | sd-cicd | elasticloadbalancing.amazonaws.com | AWS::ElasticLoadBalancing::LoadBalancer | adae1a033fe84440585e25fec3a8f3f2
> > It seems the command to create the newLB is never launching.
> 
> Looks like the service deletion never finishes. Maybe that service is in use. Are we sure no other tests are running in parallel on this cluster at the same time? I'll add some logs to debug.

@MrSantamaria  Also, the LB deleted in AWS is from another test. Our LB should be adae1a033fe84440585e25fec3a8f3f2 which isn't being deleted. I've added more logs in the output.

**Comment by ritmun** [comment]
@MrSantamaria  does the SA have required permission to delete this ELB? Just want to rule that out.

**Comment by ritmun** [comment]
Ok, I dug a bit into this. None of the aws client requests are going through. Something with the credential change is affecting this. Could you (@MrSantamaria ) look at the recent changes to creds and check if rhapi_lb.go test is affected by it?  I'm also  going to do the same.

**Comment by MrSantamaria** [comment]
Adding James as a reviewer as well since I'm looking to meet with him on this issue. 
From what I can tell we are currently running into an AWS account mismatch. The AWS Prod account I'm passing doesn't seem to be the correct account for the OSDe2e Prod jobs. 

The code is working as intended but we are sending the wrong account a request to delete the vpc.

**Review by ritmun — COMMENTED** [review]
(comment submitted while trying to close the outdated "change requested" items showing on PR. )

**Comment by ritmun** [comment]
@MrSantamaria just wanted to leave a reminder - if any more changes are needed to fix this PR, if possible, please branch off of this branch (`osd-8474-orig`) and PR back into it. It will be much quicker than fixing them on `main` branch and then pulling them into this branch.  I'll keep this branch up to date with `main`.

**Comment by MrSantamaria** [comment]
I've met with @jharrington22 on the AWS account issue we are seeing in the pr-check. 
I created this PR-Check cluster manually so it sits in a Red Hat Managed account but I will recreate it to use SD-CICD's CCS account. 

For AWS to test this fully there are three scenarios:
Manual Install(That's how I setup the pr-check cluster)
CloudCredentialOperator
CloudIngressOperator in the namespace
https://github.com/openshift/cloud-ingress-operator/blob/master/pkg/cloudclient/aws/aws.go#L110-L128

CCS Install -> This is the current logic and it will work once I make the above change. 

STS tokens -> This one might be tricky to get done.

**Comment by ritmun** [comment]
I've made a squashed PR https://github.com/openshift/osde2e/pull/1098

**Pull request closed without merging**